userspace/falco/app_cmdline_options.cpp

```diff
@@ -0,0 +1,196 @@
+	{
+		for(auto &tag : m_cmdline_parsed["t"].as<std::vector<std::string>>())
+		{
+			enabled_rule_tags.insert(tag);
+		}
+	}
+
+	// Some combinations of arguments are not allowed.
+
+	// You can't both disable and enable rules
+	if((disabled_rule_substrings.size() + disabled_rule_tags.size() > 0) &&
+	   enabled_rule_tags.size() > 0)
+	{
+		errstr = std::string("You can not specify both disabled (-D/-T) and enabled (-t) rules");
+		return false;
+	}
+
+	if (daemon && pidfilename == "") {
+		errstr = std::string("If -d is provided, a pid file must also be provided");
+		return false;
+	}
+
+	return true;
+}
+
+std::string cmdline_options::usage()
+{
+	return m_cmdline_opts.help();
+}
+
+void cmdline_options::define()
+{
+	m_cmdline_opts.add_options()
+		("h,help",                        "Print this page", cxxopts::value(help)->default_value("false"))
+		("c",                             "Configuration file. If not specified tries " FALCO_SOURCE_CONF_FILE ", " FALCO_INSTALL_CONF_FILE, cxxopts::value(conf_filename), "<path>")
+		("A",                             "Monitor all events, including those with EF_DROP_SIMPLE_CONS flag.", cxxopts::value(all_events)->default_value("false"))
+		("b,print-base64",                "Print data buffers in base64. This is useful for encoding binary data that needs to be used over media designed to.")
+		("cri",                           "Path to CRI socket for container metadata. Use the specified socket to fetch data from a CRI-compatible runtime. If not specified, uses libs default.", cxxopts::value(cri_socket_path), "<path>")
+		("d,daemon",                       "Run as a daemon.", cxxopts::value(daemon)->default_value("false"))
+		("disable-cri-async",             "Disable asynchronous CRI metadata fetching. This is useful to let the input event wait for the container metadata fetch to finish before moving forward. Async fetching, in some environments leads to empty fields for container metadata when the fetch is not fast enough to be completed asynchronously. This can have a performance penalty on your environment depending on the number of containers and the frequency at which they are created/started/stopped.", cxxopts::value(disable_cri_async)->default_value("false"))
+		("disable-source",                "Disable a specific event source. Available event sources are: syscall, k8s_audit, or any source from a configured source plugin. It can be passed multiple times. Can not disable all event sources.", cxxopts::value(disable_sources), "<event_source>")
+		("D",                             "Disable any rules with names having the substring <substring>. Can be specified multiple times. Can not be specified with -t.", cxxopts::value(disabled_rule_substrings), "<substring>")
+		("e",                             "Read the events from <events_file> (in .scap format for sinsp events, or jsonl for k8s audit events) instead of tapping into live.", cxxopts::value(trace_filename), "<events_file>")
```

Comment:
Is `jsonl` a typo?

Reply:
No, jsonl is a "real" format: https://jsonlines.org/. It's just a collection of json objects, each on their own line, separated by newlines.